public/components/model_list/model_version_undeployed_modal.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useImperativeHandle, useRef, useState } from 'react';
+import {
+  EuiButton,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+} from '@elastic/eui';
+import { bus } from './bus';
+export class NoIdProvideError {}
+export interface ModelConfirmDeleteModalInstance {
+  show: (modelId: string) => void;
+}
+let deployedVersion: string[] = [];
+export const ModelVersionUndeployedModal = React.forwardRef<
+  ModelConfirmDeleteModalInstance,
+  { onDeleted: () => void }
+>(({ onDeleted }, ref) => {
+  const deleteIdRef = useRef<string>();
+  const [visible, setVisible] = useState(false);
+  const handleCancel = useCallback(() => {
+    setVisible(false);
+    deleteIdRef.current = undefined;
+  }, []);
+  useImperativeHandle(
+    ref,
+    () => ({
+      show: (id: string) => {
+        deleteIdRef.current = id;
+        setVisible(true);
+      },
+    }),
+    []
+  );
+  function receiveVal() {
+    bus.on('sendVersions', (data) => {
+      deployedVersion = data;
+    });
+  }
+  receiveVal();
```

Comment:
We don't need this. The `receiveVal` will execute many times after render many times.